src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MemoryPoolViewBufferScope.cs

```diff
@@ -60,7 +65,7 @@ public ViewBufferValue[] GetSegment()
 
             try
             {
-                segment = _viewBufferPool.Rent(SegmentSize);
+                segment = _viewBufferPool.Rent(Math.Max(size, MinimumSize));
```

Comment:
My suggestion would be to remove the `Math.Max()` call as well as `MinimumSize`. This code isn't part of a public API.